source-code-message.php

```diff
@@ -33,5 +33,5 @@ function classloader($class) {
 spl_autoload_register(__NAMESPACE__ . '\classloader');
 
 add_action('plugins_loaded', __NAMESPACE__ . '\Plugin::loadTextdomain');
-add_action('init', __NAMESPACE__ . '\Plugin::init');
+add_action('wp', __NAMESPACE__ . '\Plugin::init');
```

Comment:
While the `init` action is the typical entry point for the hook registration, you should rename the method also if you are choosing to use the `wp` hook instead. Another possible way would be something like this:

```php
add_action('init', __NAMESPACE__ . '\Plugin::init');

// Plugin.php
class Plugin {

  public static function init() {
    add_action('wp', __CLASS__ . '::wp');
  }

  public static function wp() {
    // code goes here…
  }

}
```

I'd rather tend to use my suggested way so you can see from the init hook what's going on in the plugin. Also we are using this schema in all our plugins.